.github/workflows/develop.yml

```diff
@@ -0,0 +1,128 @@
+name: develop
+
+on:
+  pull_request:
+  push:
+
+jobs:
+
+  linux-x86:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Dep
+        run: |
+          rustup component add clippy
+          rustup component add rustfmt
+      # https://github.com/EmbarkStudios/cargo-deny-action
+      - uses: EmbarkStudios/cargo-deny-action@v1
+        with:
+          arguments: --all-features
+          command: check advisories licenses sources bans
+      - name: Run ci-generated
+        run: make ci-generated
+      - name: Run ci
+        run: make ci
+      - name: Run ci-asm
+        run: make ci-asm
+      - name: Run ci-asm-chaos
+        run: make ci-asm-chaos
```

Comment:
Personally, I would do a `make clean` or `cargo clean` in between those commands. Previously we were doing separate isolated jobs so it might be fine. But if we choose to do those tasks in a single job, I would be concerned above un-clean cargo environments.

Reply:
My intention is to avoid compiling dependencies multiple times,

If we are going to call `cargo clean` after each test, then I think it is better to separate them into independent jobs.

Reply:
I do think time is not a critical issue here, personally I feel better we have separated, isolated jobs for each tasks.